docs/src/modules/components/AppNavDrawer.js

```diff
@@ -123,28 +123,30 @@ ProductIdentifier.propTypes = {
   versionSelector: PropTypes.element,
 };
 
+const browserUrlPreviewHeight = 30;
+
 function PersistScroll(props) {
   const { slot, children, enabled } = props;
   const rootRef = React.useRef();
 
   useEnhancedEffect(() => {
-    const parent = rootRef.current ? rootRef.current.parentElement : null;
-    const activeElement = parent.querySelector('.app-drawer-active');
+    const scrollContainer = rootRef.current ? rootRef.current.parentElement : null;
+    const activeDrawerLink = scrollContainer.querySelector('.app-drawer-active');
 
-    if (!enabled || !parent || !activeElement || !activeElement.scrollIntoView) {
+    if (!enabled || !scrollContainer || !activeDrawerLink || !activeDrawerLink.scrollIntoView) {
       return undefined;
     }
 
-    parent.scrollTop = savedScrollTop[slot];
+    scrollContainer.scrollTop = savedScrollTop[slot];
 
-    const activeBox = activeElement.getBoundingClientRect();
+    const activeBox = activeDrawerLink.getBoundingClientRect();
 
-    if (activeBox.top < 0 || activeBox.top > window.innerHeight) {
-      parent.scrollTop += activeBox.top - 8 - 32;
+    if (activeBox.top < 0 || activeBox.bottom + browserUrlPreviewHeight > window.innerHeight) {
+      scrollContainer.scrollTop += activeBox.top - (activeBox.bottom - activeBox.top) * 1.5;
```

Comment:
This real fix is here, the rest is renaming variables that got me confused.

We want to test the bottom of the box, not the top
We want to also scroll if the URL preview can overlay over the active page
We want to replace these magic 8 and 32 numbers, I have no idea what they are, i.e. I don't remember.